graphics/slider.py

```diff
@@ -0,0 +1,93 @@
+import math
+from pygame import Rect, Color, Surface, transform
+
+from graphics.graphics_manager import get_font
+from graphics.outline_text import OutlineText
+
+
+class Slider:
+    value: int
+    min: int
+    max: int
+    rect: Rect
+    knob_width: int
+    slider_height: int
+    slider_surface: Surface
+    slider_color: Color
+    knob_color: Color
+    label: Surface
+    
+    _label_rect: Rect
+    _slider_rect: Rect
+    
+    def __init__(
+        self,
+        min_value: int,
+        max_value: int,
+        rect: Rect,
+        default_value: int | None = None,
+        knob_width: int | None = None,
+        slider_height: int | None = None,
+        slider_surface: Surface | None = None,
+        slider_color: Color = Color("white"),
+        knob_color: Color = Color("red"),
+        label: str | Surface = "",
+    ) -> None:
+        self.min = min_value
+        self.max = max_value
+        self.rect = rect
+        self.value = default_value if default_value is not None else min_value
+        self.knob_width = (
+            knob_width
+            if knob_width is not None
+            else min(math.floor(rect.width / 12), 20)
+        )
+        
+        self.slider_height = slider_height if slider_height < rect.height else rect.height
+        
+        if str is type(label):
```

Comment:
I think it's clearer to write
```suggestion
        if type(label) is str:
```